### PR TITLE
New tutorial: Locally Connected Microcontrollers and Real-Time Analytics (Part 1 of 2)

### PR DESCRIPTION
**Pull request by lepistom**
This part 1 of a two-part tutorial demonstrates how to use an Arduino Microcontroller to provide analog sensor readings to a Raspberry Pi], connect the devices to Google Cloud IoT Core, post telemetry data from the devices, and analyze the data in real time. This part 1 of the tutorial creates a 'hybrid' device, combining the strengths of a Linux-based microprocessor with internet connectivity and TLS stack, together with a constrained microcontroller for analog I/O. The example devices act as a cloud-connected solar and wind power generator.

### CODE REVIEW

**Comment by ryanmcginnis** [comment]
I'm sorry! I don't have cycles to review this week.

**Comment by ryanmcginnis** [comment]
Assigned to @dghale for technical writer review. (Was hoping that dpebot would auto-assign.)

File: tutorials/ardu-pi-serial-part-1/index.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+This tutorial uses billable components of GCP, including:
+
+- Cloud IoT Core
+- Cloud Pub/Sub
+
+This tutorial should not generate any usage that would not be covered by the [free tier](https://cloud.google.com/free/), but you can use the [Pricing Calculator](https://cloud.google.com/products/calculator/) to generate a cost estimate based on your projected production usage.
+
+## Required hardware
+- Raspberry Pi 3 Model [B][3b] or [B+][3b+]. **Note:** the instructions should work on other Pi models as well, but this has not been verified
```
Comment (dghale) [inline tutorials/ardu-pi-serial-part-1/index.md]: Add a period after "verified."

File: tutorials/ardu-pi-serial-part-1/index.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+![boards diagram][boardsdiag]
+[boardsdiag]: ardu-pi.png
+
+1. Connect 2 analog sensors to the Arduino. See [Figure 2][boardsdiag] to see how to wire the 2 sensors to the Arduino. In this example, we will read the analog voltage values of a DC motor, and a solar panel. *Note: you can replace the DC motor and solart panel with other analog sensors - but you **have to ensure that the sensor output does not exceed 5V** or it can damage the Arduino pin.*
+1. Connect the first sensor (small DC motor) to the Arduino. Connect the black ground wire of the motor to a `GND` port, and the red (voltage input/output) wire to port `A0` on the Arduino.
+1. Connect the second sensor (small solar panel) to the Arduino. Connect the black ground wire of the panel to a `GND` port, and the red (voltage output) wire to port `A1` on the Arduino.
+
+### Compile the Arduino Sketch
+1. Connect your Arduino to your local development machine with a USB cable. Note that the Arduino Uno board used as the example in this tutorial, has a USB type B connector on the board. The USB cable will provide both power, and serial data connectivity to the Arduino from your machine.
```
Comment (dghale) [inline tutorials/ardu-pi-serial-part-1/index.md]: Strike the comma after the word "tutorial", as well as the comma after the word "power".

File: tutorials/ardu-pi-serial-part-1/index.md [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+1. Ensure that the baud rate is set to "115200 baud".
+1. Test requesting the current sensor values from the Arduino, by typing `0` (zero) in the Serial Monitor's input window, and pressing `<enter>`. If the Sketch and serial connection works, you should see a response similar to: `S s:136 w:43` in the Serial Monitor.
+
+Here, `S` indicates sensor output, `s:136` is the current value of the solar panel (analog pin `A1`) and `w:43` is the current value of the wind generator / DC motor (analog pin `A0`). This is how the Python client app on the Pi will request the sensor readings from the Arduino later on. Command `0` is an example of a simple protocol used to command an Arduino program.
+
+Different inputs (e.g `1`) could be used to command the Arduino to perform a different action, such as turning a servo motor. In this tutorial, the Arduino Sketch implements a single command `0`, which reads the voltage values of analog pins `A0` and `A1`.
+
+- Unplug the USB cable from your local machine, to disconnect and power down the Arduino. The Arduino will now run the sketch automatically, when ever it is connected to a power source.
```
Comment (dghale) [inline tutorials/ardu-pi-serial-part-1/index.md]: Strike the comma after the word "machine".

Comment (dghale) [inline tutorials/ardu-pi-serial-part-1/index.md]: :s/when ever/whenever

File: tutorials/ardu-pi-serial-part-1/index.md [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+```
+1. Verify that your telemetry data is streaming to Cloud Pub/Sub, by executing: 
+```
+$ gcloud pubsub subscriptions pull my-device-events-sub --limit 100 --auto-ack
+```
+You should see message payloads similar to:
+```
+{"temperature": 36.62, "timestamp": "2018-07-25T08:42:10+08:00", "clientid": "raspberrypi", "humidity": 38.44, "pressure": 1017.05, "windgen": 0.1, "solargen": 1.2}   │ 151338809303544 │ deviceId=my-device
+```
+Now your Arduino and Pi are streaming telemetry data through Cloud IoT Core to Pub/Sub. You should see the sensor readings change if you expose the solar panel to more, or less light. And the wind generator (DC motor) will start generating power, as you turn the rotor.
```
Comment (dghale) [inline tutorials/ardu-pi-serial-part-1/index.md]: :s/And the/The

File: tutorials/ardu-pi-serial-part-1/index.md [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+
+## Next
+In part 2 (coming soon) of this tutorial, you will use Cloud Dataflow with a Python streaming pipeline, to process this data stream, store it in Cloud BigQuery, explore the data with a Cloud Datalab Jupyter Notebook, and visualize it using Cloud Data Studio.
```
Comment (dghale) [inline tutorials/ardu-pi-serial-part-1/index.md]: Strike the comma after the word "pipeline".

File: tutorials/ardu-pi-serial-part-1/index.md (hunk 1 above)
Comment (dghale) [inline tutorials/ardu-pi-serial-part-1/index.md]: :s/verified/verified.

File: tutorials/ardu-pi-serial-part-1/index.md [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+
+## Create an IoT Core registry and device
+First, execute all the steps in the IoT Core [Quickstart][quickstart] *on your local development environment e.g laptop*. But *do not clean up* the environment. The rest of this tutorial will use the registry, device and device keys created with the quickstart.
+
+[quickstart]: https://cloud.google.com/iot/docs/quickstart
+
+## Raspberry Pi installation and configuration
+###  Raspbian OS installation and configuration
+1. Follow the instructions [here][raspbian] to download the latest [Raspbian Lite OS image][raspbianimg], and write it on your MicroSD card. All the steps in this tutorial run in console mode on the Pi, or over SSH to the Pi.
+1. Insert your Pi in the Pi case.
```
[the remark this answers is not in the document]
Comment (lepistom) [reply tutorials/ardu-pi-serial-part-1/index.md]: Done

File: tutorials/ardu-pi-serial-part-1/index.md (hunk 2 above)
Comment (lepistom) [reply tutorials/ardu-pi-serial-part-1/index.md]: Done

File: tutorials/ardu-pi-serial-part-1/index.md (hunk 3 above)
Comment (lepistom) [reply tutorials/ardu-pi-serial-part-1/index.md]: Done

Comment (lepistom) [reply tutorials/ardu-pi-serial-part-1/index.md]: Done

File: tutorials/ardu-pi-serial-part-1/index.md (hunk 4 above)
Comment (lepistom) [reply tutorials/ardu-pi-serial-part-1/index.md]: Done

File: tutorials/ardu-pi-serial-part-1/index.md (hunk 1 above)
Comment (lepistom) [reply tutorials/ardu-pi-serial-part-1/index.md]: Done

File: tutorials/ardu-pi-serial-part-1/index.md (hunk 5 above)
Comment (lepistom) [reply tutorials/ardu-pi-serial-part-1/index.md]: Done

**Comment by ptone** [comment]
LGTM

Several suggested fixes already made pre-PR tech review

File: tutorials/ardu-pi-serial-part-1/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# How to Contribute
```
Comment (jmdobry) [inline tutorials/ardu-pi-serial-part-1/CONTRIBUTING.md]: This file can be removed.

File: tutorials/ardu-pi-serial-part-1/LICENSE [hunk 8]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (jmdobry) [inline tutorials/ardu-pi-serial-part-1/LICENSE]: This file can be removed.

File: tutorials/ardu-pi-serial-part-1/index.md [hunk 9]
```diff
@@ -0,0 +1,330 @@
+---
+title: Locally Connected Microcontrollers and Real-Time Analytics (Part 1 of 2)
+description: Learn how to integrate an Arduino with a Raspberry Pi, and connect them as one device to Google Cloud IoT Core
```
Comment (jmdobry) [inline tutorials/ardu-pi-serial-part-1/index.md]: Description needs to end with a period.

File: tutorials/ardu-pi-serial-part-1/index.md [hunk 10]
```diff
@@ -0,0 +1,330 @@
+---
+title: Locally Connected Microcontrollers and Real-Time Analytics (Part 1 of 2)
```
Comment (jmdobry) [inline tutorials/ardu-pi-serial-part-1/index.md]: For the 3 images in this file, you need to prefix with `https://storage.googleapis.com/gcp-community/tutorials/ardu-pi-serial-part-1/`

e.g. https://storage.googleapis.com/gcp-community/tutorials/ardu-pi-serial-part-1/ardu-pi.png

File: tutorials/ardu-pi-serial-part-1/CONTRIBUTING.md (hunk 7 above)
Comment (lepistom) [reply tutorials/ardu-pi-serial-part-1/CONTRIBUTING.md]: Done

File: tutorials/ardu-pi-serial-part-1/LICENSE (hunk 8 above)
Comment (lepistom) [reply tutorials/ardu-pi-serial-part-1/LICENSE]: Done

File: tutorials/ardu-pi-serial-part-1/index.md (hunk 9 above)
Comment (lepistom) [reply tutorials/ardu-pi-serial-part-1/index.md]: Done

File: tutorials/ardu-pi-serial-part-1/index.md (hunk 10 above)
Comment (lepistom) [reply tutorials/ardu-pi-serial-part-1/index.md]: Done

**Comment by jmdobry** [comment]
Published to https://cloud.google.com/community/tutorials/ardu-pi-serial-part-1

Please review and make sure it looks okay.

**Pull request merged**